KMB/make_KMB_info.py

```diff
@@ -0,0 +1,270 @@
+
+        cats = list(set(cats))  # remove any duplicates
+        return cats
+
+    # @todo: Implement creator from mapping - T164567
+    def generate_meta_cats(self, item, content_cats):
+        """
+        Produce maintanance categories related to a media file.
+
+        :param item: the metadata for the media file in question
+        :param content_cats: any content categories for the file
+        :return: list of categories (without "Category:" prefix)
+        """
+        pass
+        cats = []
+
+        # base cats
+        # "Images from the Swedish National Heritage Board" already added by
+        # Kulturmiljöbild-image template
+        cats.append(self.batch_cat)
+
+        # problem cats
+        if not content_cats:
+            cats.append(self.make_maintanance_cat('improve categories'))
+        # if not item.get_description():
+        #     cats.append(self.make_maintanance_cat('add description'))
+
+        # creator cats are classified as meta
+        # @todo: '{{safesubst:User:Lokal_Profil/nycklar/creators|%s|c}} % item.byline
+
+        cats = list(set(cats))  # remove any duplicates
+        return cats
+
+    @classmethod
+    def main(cls, *args):
+        """Command line entry-point."""
+        super(KMBInfo, cls).main(*args)
+
+
+class KMBItem(object):
+    """Store metadata and methods for a single media file."""
+
+    def __init__(self, initial_data):
+        """
+        Create a KMBItem item from a dict where each key is an attribute.
+
+        :param initial_data: dict of data to set up item with
+        """
+        # ensure all required varaibles are present
+        required_entries = ('latitude', 'longitude', 'avbildar')
+        for entry in required_entries:
+            if entry not in initial_data:
+                initial_data[entry] = None
+
+        for key, value in initial_data.iteritems():
+            setattr(self, key, value)
+
+    def get_wiki_description(self):
+        """Generate the wikitext description."""
+        wiki_descritpion = ''
+        if self.motiv != self.namn:
+            wiki_descritpion = self.motiv + ' '
+        if self.avbildar:
+            wiki_descritpion += ' '.join(self.avbildar)
+
+        return wiki_descritpion.strip()
+
+    # @todo: construct a fallback for descriptions,
+    #        and ensure meta cats tie in to this
+    def get_description(self):
+        """Construct an appropriate description."""
+        if self.namn:
+            return self.namn.replace('S:t', 'St')
+        else:
+            raise NotImplementedError
+
+    # @todo: use kommunkod/sockenkod to go via wikidata - T164576
+    def get_depicted_place(self):
+        """Get a linked version of the depicted place."""
+        s_triggers = ('a', 'o', 'u', 'å', 'e', 'i', 'y', 'ä', 'ö', 's', 'x')
+        depicted_place = None
+        if not self.land or self.land == 'se':
+            kommun_link = self.kommunName
+            # add s if kommun does not end in s or a vowel
```

Comment:
I'm completely dropping this in favour of wikidata. But in a follow up patch fullfilling [T164576](https://phabricator.wikimedia.org/T164576)